src/lib.rs

```diff
@@ -0,0 +1,451 @@
+// Copyright 2022 Contributors to the Veraison project.
+// SPDX-License-Identifier: Apache-2.0
+
+#[derive(thiserror::Error, PartialEq, Eq)]
+pub enum Error {
+    #[error("configuration error: {0}")]
+    ConfigError(String),
+    #[error("API error: {0}")]
+    ApiError(String),
+    #[error("callback error: {0}")]
+    CallbackError(String),
+    #[error("feature not implemented: {0}")]
+    NotImplementedError(String),
+}
+
+// While for other error sources the mapping may be more subtle, all reqwest
```

Comment:
```suggestion
// While for other error sources the mapping may be more subtle, all request
```

Reply:
`reqwest` is the name of the crate we are using.

Reply:
Oh OK, any reason for this specific name reqwest ?

Reply:
IDK

Reply:
Yeah, the Cargo crate namespace is global, so you can't have two crates with the name `request`, and `request` was already being squatted by someone else (either "maliciously", because they wanted to sell the name, or someone who started a library but didn't actually do a proper job with it)... :) quite common in the Rust world, unfortunately. And `reqwest` is close enough, I guess 😄